v201710/sandbox_test.go

```diff
@@ -933,3 +933,40 @@ func TestRateError(t *testing.T) {
 	wg.Wait()
 
 }
+
+func Testrequest(t *testing.T) {
+	config := getTestConfig()
+	campaigns, _, err := NewCampaignService(&config.Auth).Get(Selector{
+		Fields: []string{"Id", "Name", "CampaignId"},
+	})
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	campaignId := campaigns[0].Id
+	adGroups, err := NewAdGroupService(&config.Auth).Mutate(
+		AdGroupOperations{
+			"ADD": {
+				AdGroup{
+					Name:       "test ad group " + rand_str(10),
+					Status:     "PAUSED",
+					CampaignId: campaignId,
+				},
+			},
+		},
+	)
+
+	if err != nil && err.Error() == "" {
+		t.Fatal(err.Error())
+	}
+
+	defer func() {
```

Comment:
Put this at the beginning of the test so that it definitely gets registered no matter what. If for some reason `t.Fatal` above is called it'll never get to this func

Reply:
Made the change

Reply:
Actually put it alllll the way at the top. You moved it up a little bit but there's still another error check that could cause the test to bail before getting to your `defer` statement. 

For more information on defers, check out: https://blog.golang.org/defer-panic-and-recover

Reply:
Also, when I run this test I get this:

```
panic: runtime error: index out of range [recovered]
	panic: runtime error: index out of range

goroutine 5 [running]:
testing.tRunner.func1(0xc420070ea0)
	/usr/local/go/src/testing/testing.go:622 +0x29d
panic(0x700ea0, 0x8d8d20)
	/usr/local/go/src/runtime/panic.go:489 +0x2cf
github.com/Getsidecar/gads/v201710.TestSandboxRequest.func1(0x0, 0x0, 0x0, 0xc420121db0, 0xc420121f18, 0xc420070ea0)
	/home/d/projects/go/src/github.com/Getsidecar/gads/v201710/sandbox_test.go:961 +0x262
github.com/Getsidecar/gads/v201710.TestSandboxRequest(0xc420070ea0)
	/home/d/projects/go/src/github.com/Getsidecar/gads/v201710/sandbox_test.go:971 +0x4d1
testing.tRunner(0xc420070ea0, 0x772848)
	/usr/local/go/src/testing/testing.go:657 +0x96
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:697 +0x2ca
exit status 2
FAIL	github.com/Getsidecar/gads/v201710	0.519s
```

Because the deferred function doesn't have `adGroups` in its scope, you should pass it in as an argument